### PR TITLE
Feature: Product details

### PR DESCRIPTION
**Pull request by ksookdhew**
Screenshot 2024-08-07 at 13 42 29
Screenshot 2024-08-07 at 13 42 51
Screenshot 2024-08-07 at 13 43 00

### CODE REVIEW

**Review by AdeWet — APPROVED** [review]

File: src/features/productDetails/ProductDetails.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import {formattedPrice} from "../../utils/productUtils.ts";
+import {Link, useParams} from "react-router-dom";
+import {useQuery} from "@tanstack/react-query";
+import {getProductWithId} from "../../services/api.ts";
+
+const ProductDetails = () => {
+    const {productId} = useParams()
+    const productQuery = useQuery({queryKey: [productId], queryFn: () => getProductWithId(productId ?? '')})
+
+
```
Comment (sjacobs1) [inline src/features/productDetails/ProductDetails.tsx:10]: is it necessary to make another query for the product details?

**Comment by sjacobs1** [comment]
I just noticed that on your screenshot the price for Men's casual slim fit T-Shirt is 22.00, but price on API is 22.30(22.3). just check the rounding maybe since it's changing the price value.

File: src/features/productDetails/ProductDetails.tsx (hunk 1 above)
Comment (ksookdhew) [reply src/features/productDetails/ProductDetails.tsx:10]: I'm actually not sure if it's completely necessary, this was my solution to navigate with the router to the product details page. If you think there is a better way I would greatly appreciate any suggestion 😄

Comment (sjacobs1) [reply src/features/productDetails/ProductDetails.tsx:10]: all good, was just a genuine thought. I would have used selected state to get the selected product id and display its details. really not sure about the best way, just a thought 😃

**Review by sjacobs1 — APPROVED** [review]

File: src/features/productDetails/ProductDetails.tsx (hunk 1 above)
Comment (ksookdhew) [reply src/features/productDetails/ProductDetails.tsx:10]: Yeah, definitely should have a discussion about it next time

**Pull request merged**